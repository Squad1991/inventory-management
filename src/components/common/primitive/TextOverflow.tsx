import Tooltip from 'rc-tooltip';
import React from 'react';
import { ComponentPropsWithTestId } from '~src/components/common/utils/types';

interface TextOverflowProps extends ComponentPropsWithTestId {
  text: string;
}

const TextOverflow: React.FC<TextOverflowProps> = ({ text, testId }) => {
  return (
    <Tooltip mouseEnterDelay={0.5} placement="right" trigger={['hover']} overlay={text}>
      <div
        data-testid={testId}
        style={{ overflow: 'hidden', textOverflow: 'ellipsis', whiteSpace: 'nowrap' }}
      >
        {text}
      </div>
    </Tooltip>
  );
};

export default TextOverflow;
