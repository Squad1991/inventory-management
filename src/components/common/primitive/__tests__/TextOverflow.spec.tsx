import { render } from '@testing-library/react';
import React from 'react';
import TextOverflow from '~src/components/common/primitive/TextOverflow';

describe('TextOverflow', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <div style={{ width: '1px' }}>
        <TextOverflow text="Lorem ipsum dolor sit ametLorem ipsum dolor sit amet" />
      </div>,
    );
    expect(container).toMatchSnapshot();
  });
});
