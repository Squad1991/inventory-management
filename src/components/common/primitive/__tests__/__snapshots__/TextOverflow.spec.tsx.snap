// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TextOverflow > should match snapshot 1`] = `
<div>
  <div
    style="width: 1px;"
  >
    <div
      style="overflow: hidden; text-overflow: ellipsis; white-space: nowrap;"
    >
      Lorem ipsum dolor sit ametLorem ipsum dolor sit amet
    </div>
  </div>
</div>
`;
